### PR TITLE
Native lib rework part 1

### PR DESCRIPTION
**Pull request by Simn**
This PR starts the unification of native lib handling. To that end, we OOP a bit to introduce a `native_library` class and then extend it for various target-specific libs. I know nobody likes OCaml OOP but this should be a good use-case for it.

Ideally, this shouldn't change behavior at all. However, we have virtually no test coverage beyond really basic setup, so I would appreciate if people could test this a bit (@nadako for swf, @kLabz for C#... nobody uses Java). I'll probably force you to test this by merging soon anyway~

The next step is to make the compilation server aware of native libraries.

### CODE REVIEW

File: src/codegen/dotnet.ml [hunk 1]
```diff
@@ -1204,14 +1203,28 @@ let add_net_lib com file std =
...
+	initializer
+		if std then self#add_flag FlagIsStd
+end
+
+let add_net_lib com file std =
+	let real_file = if Sys.file_exists file then
+		file
+	else try Common.find_file com file with
+		| Not_found -> try Common.find_file com (file ^ ".dll") with
+		| Not_found ->
```
Comment (RealyUniqueName) [inline src/codegen/dotnet.ml:1219]: This is hard to read. At first sight I got "Why two `Not_found` cases"?

Comment (Simn) [reply src/codegen/dotnet.ml:1219]: I just copied that code. Feel free to clean up!

**Comment by kLabz** [comment]
I can't test this week, only next week.

**Comment by nadako** [comment]
will compile our flash things tomorrow at work :D

**Pull request merged**